docs/Practices/hub.md

```diff
@@ -0,0 +1,80 @@
+# Working with Bounded Contexts in Hub
+
+## Intro
+We utilize a combination of Domain Driven Design, Bounded Contexts, Command Query Separation, Verticle Slices Architecture, Functional styled Procedural Programming, Ports and Adapters, The Repository Pattern, and Onion (aka Clean aka Hexagonal) Architecture within backend code here at Prenda. As of this writing these patterns are only really used for new code within the Hub application. For the puroses of this document we'll reference these and other core patterns, paradigms, tools, or principles as "useful abstractions (UA)". 
+
+The purpose of utilizing these UAs is to help achieve a set of defined goals that are outlined further in this document.
+
+Not all UAs are implemented dogmatically, and doing so would lead to many instances of conflict between the UAs. Rather choice core principles from the UAs have been chosen and applied in such a way that these principles are orthogonal to each other and support well the defined goals.
+
+This document will not attempt to answer all critiques of these UAs nor will it discuss the in depth the tradeoffs of why these UAs are the right tools.
+
+## Goals
+- Code for a given capability is easy to find.
+- Changes to a given behavior are less risky.
+- The effects of a change to a domain model shape are clear and should cascade and only to those dependents that actually utilize the changes.
+- The API of code implementing a behavior is clear. *You should know exactly what data is required to successfully initiate a behavior as well as understanding exactly how 
+- Failures due to essential complexity should be uniquely and easily identifiable and bad data states caused by such errors should be easily remediated.
+- Code should be covered by automated tests. *Testing strategy and acceptable coverage (shoot for as much as possible) are not currently specified).
+
+## Non Goals
+- Don't Repeat Yourself (DRY) and Brevity at the expensive of clarity. *DRYness and Brevity are often associated with readability and understandability, but these should be seen as side effects of Clarity. If ever in doubt about a decision between dry/brevity and clarity, err on the side of clarity. Flexibility is also valuable given Prenda's current state. There are many instances in which sacrificing DRYness and Brevity for flexibility is the correct choice.*
+
+## Guidelines
+- Other than the consumption of a command or query, code sharing between bounded contexts should be avoided by default. There may be cases where a bounded context is implemented in a different project. In this case no dependency should be had on the code of that project and integration must be done asynchronously (http call, event handler, etc).  *This is a concept borrowed from Domain Driven Design - specifically Bounded Contexts.*
```

Comment:
It would be awesome if we made these collapsible headers with examples, or links to examples

Reply:
guidelines now have links to useful abstractions - working through examples as applicable.